### DISCUSSION TITLE
RTL Support

### DISCUSSION THREAD

**Discussion by sambeta01** [absadiki/whatsapp-msgstore-viewer#8] [Ideas]
To support RTL languages, Kivy must use Pango to handle the text, to achieve that:
1- Install [GTK3 Runtime](https://github.com/tschoonj/gtk-for-windows-runtime-environment-installer/releases)
2- pip install pango
3- add the following line to the main.py file (after `import os`):
`os.environ['KIVY_TEXT'] = 'pango'`

**Comment by absadiki** [comment]
@sambeta01, Glad to hear that you got RTL working. 
I'll add a note in the README for this in case someone faces the same issue. 
Thanks!

**Reply by sambeta01** [reply]
You could add pango to the requirements, and the line to main.py. 
Then add the note to install GTK3 for whoever needs RTL.
Thank you.

**Reply by absadiki** [reply]
Pango is still not the default way to render text in Kivy yet, so for anyone using an LTR language, SDL2 or PIL is a better option. Also, the [pango python package](https://pypi.org/project/pango) seems empty, so I doubt it has any effect. It's probably better to stick with the defaults until the official release of Kivy with Pango support.
In the meantime, I've added a note in the README linking to this thread in case someone faces a similar issue."

**Reply by sambeta01** [reply]
You are right, I didn't notice that pango package is empy!
I removed it, and removed the added line in the code, and RTL is still OK. Apparently just installing GTK3 Runtime did the trick.

**Reply by absadiki** [reply]
Yes, probably GTK3 runtime is the missing part to make it work on windows. 
Hopefully kivy 3.0 will support all languages out of the box.

**Comment by fhry** [comment]
![Screenshot 2025-01-27 131318](https://github.com/user-attachments/assets/faf2765b-c839-44a2-9127-323714f0b1b4)

I don't understand where it goes wrong, I input the right file database but still can't be opened.

any suggest?

**Reply by absadiki** [reply]
If the database was decrypted successfully, it's likely that your database schema differs from the one I have. I explained in the README how to update the schema and submit a PR, but unfortunately, there have been no contributions from the community yet in that regard.

**Reply by fhry** [reply]
Ah i see, I will try another way. I can't opened it legally because I lost my android data.
